### PR TITLE
Fix for

### PR DESCRIPTION
**Pull request by nagasakinuclear**

### CODE REVIEW

File: src/Templator/Templator.Logic/SourceFileGeneration/Contracts/IModuleFileGenerator.cs [hunk 1]
```diff
@@ -1,9 +1,11 @@
...
+﻿using Templator.Domain.DataStructures;
+using Templator.Domain.Models.TemplateConfiguration;
+using Templator.Domain.TemplateReader;
 
 namespace Templator.Logic.SourceFileGeneration.Contracts
 {
     public interface IModuleFileGenerator
     {
-        Task Generate(string path, ModuleFile module);
+        Task Generate(string path, ModuleFile module, TrieNode<ConfiguredTemplate> configuredTemplates);
```
Comment (nagasakinuclear) [inline src/Templator/Templator.Logic/SourceFileGeneration/Contracts/IModuleFileGenerator.cs:9]: TODO: pass forest of configurations into the ModuleFileGenerator

File: src/Templator/Templator.Logic/SourceFileGeneration/Contracts/IModuleGeneratorStrategy.cs [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
+using Templator.Domain.DataStructures;
+using Templator.Domain.Models.TemplateConfiguration;
 using Templator.Domain.TemplateReader;
 
 namespace Templator.Logic.SourceFileGeneration.Contracts
 {
     public interface IModuleGeneratorStrategy
     {
-        Task Generate(ModuleFile module, string path);
+        Task Generate(ModuleFile module, string path, TrieNode<ConfiguredTemplate> configuredTemplates);
```
Comment (nagasakinuclear) [inline src/Templator/Templator.Logic/SourceFileGeneration/Contracts/IModuleGeneratorStrategy.cs:10]: TODO: pass forest of configurations into the ModuleFileGenerator

File: src/Templator/Templator.Logic/SourceFileGeneration/TemplateFileContentGenerator.cs [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+using Templator.Domain.TemplateReader;
+using Templator.Logic.SourceFileGeneration.CodeGenerator.Contracts;
+using Templator.Logic.SourceFileGeneration.Contracts;
+using Templator.Logic.SourceFileGeneration.CSharp;
+
+namespace Templator.Logic.SourceFileGeneration
+{
+    public class TemplateFileContentGenerator : ITemplateFileContentGenerator
+    {
+        private readonly ITemplateOutputFileGenerator _templateOutputFileGenerator;
```
Comment (nagasakinuclear) [inline src/Templator/Templator.Logic/SourceFileGeneration/TemplateFileContentGenerator.cs:15]: TODO: rename ITemplateOutputFileGenerator to ITemplateOutputGenerator

File: src/Templator/Templator.Logic/SourceFileGeneration/TemplateFileContentGenerator.cs [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+                var content = _templateOutputFileGenerator.GenerateTemplateOutputText(configuredTemplateNode, fileToGenerateToken).ToString();
+
+                var relativePath = configuredTemplateNode.Value.RelativePath ?? (fileToGenerateToken.Value as FileToGenerateModel).DefaultRelativePath;
+
+                if (string.IsNullOrWhiteSpace(relativePath))
+                {
+                    throw new Exception("Path was not provided for " + templateModel.TemplateNameBlock + " Template");
+                }
+
+                // TODO: support multiple template extension names
```
Comment (nagasakinuclear) [inline src/Templator/Templator.Logic/SourceFileGeneration/TemplateFileContentGenerator.cs:40]: // TODO: support multiple template extension names

                // TODO: support different file names

                // TODO: do this part in CSharpProjectGenerator and also remove projectName and path parametr

File: src/Templator/Templator.Logic/SourceFileGeneration/TemplateOutputFileGenerator.cs [hunk 5]
```diff
@@ -45,7 +46,7 @@ public ReadOnlySpan<char> GenerateTemplateText(TrieNode<ConfiguredTemplate> conf
                 TemplateModel = templateModel,
                 Variables = variables
             };
-            HandleNode(templateModel.TokenTrie[i], context, builder);
+            HandleNode(templateModel.TokenTrie[i], context, builder, templateModel);
```
Comment (nagasakinuclear) [inline src/Templator/Templator.Logic/SourceFileGeneration/TemplateOutputFileGenerator.cs:49]: TODO: remove template model from params. It exists in the context

File: src/Templator/Templator.Logic/TemplateRetrieverService.cs [hunk 6]
```diff
@@ -5,23 +5,20 @@
 using Templator.Logic.DependencyGraph;
 using Templator.Logic.DependencyGraph.Alghoritms;
 using Templator.Logic.TemplateReader.Contracts;
-using Templator.Logic.TokenProcessing;
-
+ 
 namespace Templator.Logic
 {
-    public class TemplateRetrieverService : ITemplateRetrieverService
+    public class TemplateGraphBuilderService : ITemplateGraphBuilderService
```
Comment (nagasakinuclear) [inline src/Templator/Templator.Logic/TemplateRetrieverService.cs:11]: TODO: rename to TemplateGraphBuilder

File: src/Templator/Templator.Domain/Models/TemplateElements/Structural/FileToGenerateModel.cs [hunk 7]
```diff
@@ -5,6 +5,7 @@ public class FileToGenerateModel : NestingCodeBlockModel
         public string? DefaultRelativePath { get; set; }
         public int RoundBracketStartIndex { get; set; }
         public int RoundBracketEndIndex { get; set; }
+        public string DefaultExtension { get; set; } = ".cs";
```
Comment (yyasun) [inline src/Templator/Templator.Domain/Models/TemplateElements/Structural/FileToGenerateModel.cs:8]: bruh?

File: src/Templator/Templator.Logic/SourceFileGeneration/FileCreator.cs [hunk 8]
```diff
@@ -3,7 +3,7 @@ namespace Templator.Logic.SourceFileGeneration
 {
     public class FileCreator
     {
-        public async Task CreateFile(string path, string content)
+        public static async Task CreateFile(string path, string content)
```
Comment (yyasun) [inline src/Templator/Templator.Logic/SourceFileGeneration/FileCreator.cs:6]: please make thiss FileCreator Mockable

File: src/Templator/Templator.Logic/SourceFileGeneration/TemplateFileContentGenerator.cs [hunk 9]
```diff
@@ -6,69 +6,47 @@
...
 
-        public TemplateFileContentGenerator(ITemplateOutputFileGenerator templateOutputFileGenerator, ICmd cmd)
+        public TemplateFileContentGenerator(ITemplateOutputFileGenerator templateOutputFileGenerator)
         {
             _templateOutputFileGenerator = templateOutputFileGenerator;
-            _cmd = cmd;
         }
 
-        public async Task GenerateFileContent(string projectName, string path, TemplateModel templateModel, TrieNode<ConfiguredTemplate> configuredTemplateNode)
+        public IEnumerable<TemplateFileContentGeneratorOutput> GenerateFileContent(TemplateModel templateModel, TrieNode<ConfiguredTemplate> configuredTemplateNode)
```
Comment (yyasun) [inline src/Templator/Templator.Logic/SourceFileGeneration/TemplateFileContentGenerator.cs:21]: please rename GenerateFileContent to plural

**Review by yyasun — CHANGES_REQUESTED** [review]

**Pull request merged**